azurerm/resource_arm_storage_account.go

```diff
@@ -1406,13 +1406,14 @@ func expandQueuePropertiesCors(input []interface{}) *queues.Cors {
 }
 
 func flattenStorageAccountNetworkRules(input *storage.NetworkRuleSet) []interface{} {
-	if len(*input.IPRules) == 0 && len(*input.VirtualNetworkRules) == 0 {
-		return []interface{}{}
-	}
 	networkRules := make(map[string]interface{})
 
-	networkRules["ip_rules"] = schema.NewSet(schema.HashString, flattenStorageAccountIPRules(input.IPRules))
-	networkRules["virtual_network_subnet_ids"] = schema.NewSet(schema.HashString, flattenStorageAccountVirtualNetworks(input.VirtualNetworkRules))
+	if len(*input.IPRules) > 0 {
+		networkRules["ip_rules"] = schema.NewSet(schema.HashString, flattenStorageAccountIPRules(input.IPRules))
+	}
```

Comment:
can we remove this `if` check here? the flatten function will return an empty array when this is nil - which'll ensure we track diff's here:

```suggestion
	networkRules["ip_rules"] = schema.NewSet(schema.HashString, flattenStorageAccountIPRules(input.IPRules))
```